src/bin/ion/commands/beta/generate/templates/rust/struct.templ

```diff
@@ -29,12 +24,12 @@ impl {{ target_kind_name }} {
     pub fn read_from(reader: &mut Reader) -> IonResult<Self> {
         let mut abstract_data_type = {{ target_kind_name }}::default();
         {% if abstract_data_type == "Value"%}
-            abstract_data_type.value = {% if target_kind_name | is_built_in_type == false %}
-                                            {{ target_kind_name }}::read_from(reader)?;
+            abstract_data_type.value = {% if fields[0].value | is_built_in_type == false %}
+                                            {{ fields[0].value }}::read_from(reader)?;
                                         {% else %}
-                                            reader.read_{{ target_kind_name }}()?;
+                                            reader.read_{% if fields[0].isl_type_name == "symbol" %}symbol()?.text().unwrap(){% else %}{{ fields[0].value | replace(from="string", to ="str") }}()?{% endif %}{% if fields[0].value | lower == "string" %} .to_string() {% endif %};
```

Comment:
(🗺️ PR tour) Replaces `to_string()?` -> `to_str()?.to_string()` and uses `isl_type_name` to read `symbol`.